brainpad/courses/vb/variables_and_nested_statements.md

```diff
@@ -1 +1,472 @@
+```
+## Exercise
+Add a variable that starts at 0 and then it is incremented by 1 in every loop. Once the variable reaches 1000, turn the LightBulb YELLOW. Print the variable value on the output window and make sure its incrementing, then verify the LightBulb turns YELLOW when the variable go over 1000.
+
+## Inspecting Variables
+Printing the variable value worked well for small programs with a single variable. It is not practical to continuously print every variable out. Instead, we can use the debugger feature to pause the code and inspect variables.
+```
+Namespace Level2_CrashCourse
+    Public Module Module1
+
+        Sub Main()
+            Dim count As Integer
+            count = 0
+
+            While BrainPad.LOOPING
+                count = count + 1
+
+                BrainPad.DebugPrint(count)
+
+                BrainPad.Wait.Seconds(0.2)
+            End While
+        End Sub
+
+    End Module
+End Namespace
+```
+Add a breakpoint in the loop in the previous program and step through code. Move the mouse over the count variable and wait a second. A pop up window will come up showing the value of the count variable.
+
+![Write Debug Message Breakpoint](images/variables_and_nested_statements/WriteDebugMessageBreakpoint.png) 
+
+The watch windows can be used with important variables that need to be inspected regularly. Locate the watch window and type the variable name in the left sell. This will show the variable value and type.
+
+![Watch window](images/variables_and_nested_statements/Watchwindow.png)
+
+## Exercise
+Run the previous program for few loops, then pause the program, or add a breakpoint. Inspect the value on the watch window. It should be something not 0. Change the value manually to 0 and then rerun the program and observe how the output window now shows the variable going back counting from 0.
+
+## Double Variable
+While an Integer can hold very large numbers, it can�t hold fractions. For example, there is no way to set an integer to 0.5 or 123.8. The variable type Double should be used whenever fractions are needed. But wait, since a Double type variable can hold large numbers and can hold fractions, why not just use it for everything? You certainly can but Double-type variables requires more memory and requires more processing. Also, fractions are not always desirable in the programming world.
+
+```
+Class Program
+
+    Dim frequency As Double
+    Dim increment As Integer
+
+    Public Sub BrainPadSetup()
+        frequency = 0
+        increment = 0
+    End Sub
+
+    Public Sub BrainPadLoop()
+        If BrainPad.Button.IsUpPressed() Then
+            increment = 100
+        End If
+        If BrainPad.Button.IsDownPressed() Then
+            increment = -100
+        End If
+
+        If increment <> 0 Then
+            frequency = frequency + increment
+            increment = 0
+            BrainPad.Buzzer.Start(CType(frequency, Integer))
+            BrainPad.WriteDebugMessage(frequency.ToString())
```

Comment:
WriteDebugMessage doesn't exist anymore. Replaced with WriteMessageOnComputer or similar. It has an integer overload so .ToString() isn't needed.